components/camel-huaweicloud-smn/src/test/java/org/apache/camel/component/huaweicloud/smn/PublishTemplatedMessageTest.java

```diff
@@ -0,0 +1,129 @@
+            Mockito.when(SimpleNotificationUtils.resolveSmnServiceEndpoint("unit-test"))
+                    .thenReturn("http://localhost:" + wiremockServerPort);
+        } catch (MockitoException e) {
+            LOGGER.info("Mock already registered. Using existing registration");
+        }
+    }
+
+    @Test
+    public void testTemplatedNotificationSend() throws Exception {
+        boolean isMockedServerTest = testConfiguration.getProperty("region").equals("unit-test");
+        if (isMockedServerTest) {
+            LOGGER.info("region is unit-test. Starting up wiremock stubs");
+            initWireMock();
+            setupSimpleNotificationsUtilsMock();
+        }
+
+        MockEndpoint mock = getMockEndpoint("mock:publish_templated_message_result");
+        mock.expectedMinimumMessageCount(1);
+        template.sendBody("direct:publish_templated_message", null);
+        Exchange responseExchange = mock.getExchanges().get(0);
+
+        mock.assertIsSatisfied();
+
+        Assert.assertNotNull(responseExchange.getProperty("CamelSmnMesssageId"));
+        Assert.assertNotNull(responseExchange.getProperty("CamelSmnRequestId"));
+        Assert.assertTrue(responseExchange.getProperty("CamelSmnMesssageId").toString().length() > 0);
+        Assert.assertTrue(responseExchange.getProperty("CamelSmnRequestId").toString().length() > 0);
+
+        if (isMockedServerTest) {
+            Assert.assertEquals("bf94b63a5dfb475994d3ac34664e24f2", responseExchange.getProperty("CamelSmnMesssageId"));
+            Assert.assertEquals("6a63a18b8bab40ffb71ebd9cb80d0085", responseExchange.getProperty("CamelSmnRequestId"));
+
+            LoggedRequest loggedRequest = TestUtils.retrieveTemplatedNotificationRequest(getAllServeEvents());
+            LOGGER.info("Verifying wiremock request");
+            Assert.assertEquals(
+                    "http://localhost:8080/v2/9071a38e7f6a4ba7b7bcbeb7d4ea6efc/notifications/topics/urn:smn:unit-test:9071a38e7f6a4ba7b7bcbeb7d4ea6efc:reji-test/publish",
+                    loggedRequest.getAbsoluteUrl());
+            Assert.assertEquals(
+                    "eyJzdWJqZWN0IjoiVGhpcyBpcyBteSBzdWJqZWN0bGluZSIsIm1lc3NhZ2VfdGVtcGxhdGVfbmFtZSI6ImhlbGxvLXRlbXBsYXRlIiwidGFncyI6eyJwaG9uZSI6IjEyMzQ1Njc4OTAiLCJuYW1lIjoicmVqaSJ9LCJ0aW1lX3RvX2xpdmUiOiI2MCJ9",
+                    loggedRequest.getBodyAsBase64());
+            Assert.assertEquals(
+                    "{\"subject\":\"This is my subjectline\",\"message_template_name\":\"hello-template\",\"tags\":{\"phone\":\"1234567890\",\"name\":\"reji\"},\"time_to_live\":\"60\"}",
+                    loggedRequest.getBodyAsString());
+        }
+    }
+
+    private void initWireMock() throws Exception {
+        try {
+            wireMockServer = new WireMockServer(wiremockServerPort);
+            wireMockServer.start();
+        } catch (Exception e) {
+            LOGGER.info("wiremock server already registered in test context. using the same");
```

Comment:
We should throw the exception beside printing a log.

Reply:
removed usage of wiremock totally. instead using mock for SmnClient